File: domain/src/main/java/com/insurtech/kanguro/domain/model/PetPolicy.kt
package com.insurtech.kanguro.domain.model

import android.os.Parcelable
import com.insurtech.kanguro.common.enums.PolicyStatus
import kotlinx.parcelize.Parcelize
import java.util.Date

@Parcelize
data class PetPolicy(
    val deductible: Deductible? = null,
    val sumInsured: AmountInfo? = null,
    val payment: Payment? = null,
    val pet: Pet? = null,
    val preventive: Boolean? = null,
    val waitingPeriod: Date? = null,
    val waitingPeriodRemainingDays: Int = 0,
    val id: String? = null,
    val reimbursement: Reimbursement? = null,
    val startDate: Date? = null,
    val endDate: Date? = null,
    val status: PolicyStatus? = null,
    val policyExternalId: Int? = null,
    val policyOfferId: Int? = null,
    val isFuture: Boolean? = null
) : Parcelable

File: iosApp/Modules/KanguroPet/KanguroPetDomain/Sources/KanguroPetDomain/Models/PetPolicy.swift
import Foundation
import KanguroSharedDomain

public final class PetPolicy: PetPolicyProtocol {
    
    public var petId: Int?
    public var id: String?
    public var startDate: Date?
    public var waitingPeriod: Date?
    public var waitingPeriodRemainingDays: Double?
    public var endDate: Date?
    public var deductable: Deductable?
    public var reimbursment: Reimbursment?
    public var sumInsured: SumInsured?
    public var status: PolicyStatus?
    public var feedbackRate: Int?
    public var pet: Pet
    public var payment: Payment?
    public var preventive: Bool?
    public var policyExternalId: Int?
    public var policyOfferId: Int?
    public var isFuture: Bool?

    public init(
        id: String? = nil,
        startDate: Date? = nil,
        waitingPeriod: Date? = nil,
        waitingPeriodRemainingDays: Double? = nil,
        endDate: Date? = nil,
        deductable: Deductable? = nil,
        reimbursment: Reimbursment? = nil,
        sumInsured: SumInsured? = nil,
        status: PolicyStatus? = nil,
        feedbackRate: Int? = nil,
        pet: Pet,
        payment: Payment? = nil,
        preventive: Bool? = nil,
        policyExternalId: Int? = nil,
        policyOfferId: Int? = nil,
        isFuture: Bool? = nil
    ) {
        self.id = id
        self.startDate = startDate
        self.waitingPeriod = waitingPeriod
        self.waitingPeriodRemainingDays = waitingPeriodRemainingDays
        self.endDate = endDate
        self.deductable = deductable
        self.reimbursment = reimbursment
        self.sumInsured = sumInsured
        self.status = status
        self.feedbackRate = feedbackRate
        self.pet = pet
        self.payment = payment
        self.preventive = preventive
        self.policyExternalId = policyExternalId
        self.policyOfferId = policyOfferId
        self.isFuture = isFuture
    }
}

extension PetPolicy: Hashable {
    public static func == (lhs: PetPolicy, rhs: PetPolicy) -> Bool {
        return lhs.id == rhs.id
    }

    public func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}
